migration-engine/connectors/sql-migration-connector/src/sql_renderer/postgres_renderer.rs

```diff
@@ -1005,21 +1005,31 @@ fn render_postgres_alter_enum(alter_enum: &AlterEnum, schemas: Pair<&SqlSchema>)
             let columns = schemas.walk(Pair::new(*prev_colidx, next_colidx));
             let table_name = columns.previous.table().name();
             let column_name = columns.previous.name();
-            let default_str = columns
-                    .next
-                    .default()
-                    .and_then(|default| default.as_value())
-                    .and_then(|value| value.as_enum_value())
-                    .expect("We should only be setting a changed default if there was one on the previous schema and in the next with the same enum.");
 
-            let set_default = format!(
-                "ALTER TABLE {table_name} ALTER COLUMN {column_name} SET DEFAULT '{default}'",
+            // default_str_option may still be `None`, e.g. when `column_name` is a list of enum types
+            // with the `@default([])` attribute.
+            let default_str_option = columns
+                .next
+                .default()
+                .and_then(|default| default.as_value())
+                .and_then(|value| value.as_enum_value());
+
+            let mut alter_sql = format!(
+                "ALTER TABLE {table_name} ALTER COLUMN {column_name}",
                 table_name = Quoted::postgres_ident(&table_name),
                 column_name = Quoted::postgres_ident(&column_name),
-                default = escape_string_literal(default_str),
             );
 
-            stmts.push(set_default);
+            // we only set a default if there is one
+            if let Some(default_str) = default_str_option {
+                alter_sql = format!(
+                    "{} SET DEFAULT '{default}'",
+                    alter_sql,
+                    default = escape_string_literal(default_str),
+                );
+            }
```

Comment:
I think the fix can be better expressed as a single line (a .filter() after line 1003). I'd rather we never produce `ALTER TABLE ... ALTER COLUMN ...` without the last part of the clause. It's most likely a no-op.

Reply:
Do you mean something like:

```rust
    default_str_option.filter(|default_str| {
        let set_default = format!(
            "ALTER TABLE {table_name} ALTER COLUMN {column_name} SET DEFAULT '{default}'",
            table_name = Quoted::postgres_ident(&table_name),
            column_name = Quoted::postgres_ident(&column_name),
            default = escape_string_literal(default_str),
        );
        stmts.push(set_default)
    });
```

Is it idiomatic in Rust to have mutable operations inside functional methods like `.filter()` and `.map()`?

Reply:
We can also express the same logic with:

```rust
    if let Some(default_str) = default_str_option {
        let set_default = format!(
            "ALTER TABLE {table_name} ALTER COLUMN {column_name} SET DEFAULT '{default}'",
            table_name = Quoted::postgres_ident(&table_name),
            column_name = Quoted::postgres_ident(&column_name),
            default = escape_string_literal(default_str),
        );
        stmts.push(set_default)
    }
```

Reply:
No usually we avoid mutation inside iterator methods.

I meant at line 1003, insert one more call `.filter(|(_, next)| next.default.is_some())`, and that would be the only change needed in this file. If you want to also remove the unwrap in the loop body, you can `.filter_map(|(previous, next)| next.default.map(|d| ((previous, (next, d))))`.